bot/utils/regex.py

```diff
@@ -10,3 +10,5 @@
     r"([a-zA-Z0-9\-]+)",                              # the invite code itself
     flags=re.IGNORECASE
 )
+
+END_PUNCTUATION_RE = re.compile("^.+?[.?!]$")
```

Comment:
I'm not requiring it, but if it were me I would just have this constant in the file where it's used, since it only serves one function currently. It might be generic enough to be useful in the future for something else, but not right now.